zanata-war/src/main/java/org/zanata/service/impl/TranslationServiceImpl.java

```diff
@@ -186,6 +190,9 @@
         validateReviewPermissionIfApplicable(translationRequests,
                 projectIteration, hLocale);
 
+        Map<DocumentLocaleKey, List<TextFlowTargetState>> eventMap =
+            Maps.newHashMap();
```

Comment:
In fact, we shouldn't even need a map here. Within this method, every value in DocumentLocaleKey is a constant, so we should only need a single TextFlowTargetStateEvent.